includes/Story_Post_Type.php

```diff
@@ -217,6 +216,27 @@ public function init() {
 		add_filter( 'site_option_upload_filetypes', [ $this, 'filter_list_of_allowed_filetypes' ] );
 	}
 
+	/**
+	 * Activate the service.
+	 *
+	 * @param bool $network_wide Whether the activation was done network-wide.
+	 * @return void
+	 */
+	public function activate( $network_wide ) {
+		$this->add_caps_to_roles();
+	}
+
+	/**
+	 * Deactivate the service.
+	 *
+	 * @param bool $network_wide Whether the deactivation was done network-wide.
+	 * @return void
+	 */
+	public function deactivate( $network_wide ) {
+		$this->remove_caps_from_roles();
+		unregister_post_type( self::POST_TYPE_SLUG );
```

Comment:
This is probably not needed, and could lead to problems in other parts of the deactivation routine, if they somehow interact with the post type for clean-up or so.
```suggestion
```

Reply:
This is how it worked before and I am not going to make functional changes, just refactor the code. It must have been needed at some, which is why that is there. I am not going to second guess this for now.

Reply:
IIRC we need to unregister the post type before we can flush the permalinks. Otherwise the rewrites for that post type won't be removed.